torchmetrics/detection/map.py

```diff
@@ -329,12 +354,16 @@ def update(self, preds: List[Dict[str, Tensor]], target: List[Dict[str, Tensor]]
             self.detection_boxes.append(boxes)
             self.detection_labels.append(item["labels"])
             self.detection_scores.append(item["scores"])
+            if "masks" in item:
```

Comment:
don't query the item only. Also rely on the specified type in `self.iou_type` to be more memory effficient